File: S2-Regression/Pt5-DecisionTreeRegression/DecisionTreeRegression.R
### Decision Tree Regression

## Set current working directory
getwd()
setwd('C:\\Users\\rzl\\Documents\\Github\\Machine-Learning\\S2-Regression\\Pt5-DecisionTreeRegression')
getwd()

## Import data
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

## Fitting the Decision Tree Regression to the dataset
install.packages('rpart')
library(rpart)

regressor = rpart(formula=Salary~., data=dataset, control=rpart.control(minsplit=1))

## Visualizing the Decision Tree results
library(ggplot2)
x_grid = seq(min(dataset$Salary), max(dataset$Salary), 0.001)
ggplot() +
  geom_point(aes(x=dataset$Level, y=dataset$Salary), color='red') +
  geom_line(aes(x=dataset$Level, y=predict(regressor, newdata=dataset)), color='blue') +
  ggtitle('Truth or Bluff (Decision Tree Regression') +
  xlab('Level') +
  ylab('Salary')

## Predicting a new result
y_prediction = predict(regressor, data.frame(Level=6.5))

File: S2-Regression/Pt5-DecisionTreeRegression/decision_tree_regression.py
### Decision Tree Regression

## Import all the library that needed
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Import dataset with pandas
dataset = pd.read_csv('Position_Salaries.csv')

# pandas.DataFrame.iloc >> Purely integer-location based indexing for selection by position.
x = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

## Fitting Decision Tree Regression to the dataset
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state=0)
regressor.fit(x, y)

## Visualizing the Decision Tree results
x_grid = np.arange(min(x), max(x), 0.001)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x, y, color='red')
plt.plot(x_grid, regressor.predict(x_grid), color='blue')
plt.title('Truth or Bluff (Decision Tree Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

## Predicting a new result
y_prediction = regressor.predict([[6.5]])
